### PR TITLE
Assignment4: Chris Cornell

### PR DESCRIPTION
**Pull request by ccornell88**

### CODE REVIEW

**Comment by kflannery2001** [comment]
I will review this

File: Logger/Book.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Logger;
+
+public record Book(string Title) : BaseEntity
```
Comment (kflannery2001) [inline Logger/Book.cs:3]: DO use record class (C# 10.0) for clarity, rather than the abbreviated record-only syntax.

File: Logger/Employee.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Logger;
+
+public record Employee(FullName FullName) : BaseEntity
```
Comment (kflannery2001) [inline Logger/Employee.cs:3]: record class

File: Logger/Student.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Logger;
+
+public record Student(FullName FullName) : BaseEntity
```
Comment (kflannery2001) [inline Logger/Student.cs:3]: record class

File: Logger/Book.cs (hunk 1 above)
Comment (ccornell88) [reply Logger/Book.cs:3]: Ill get that changed. I am not quite done yet just wanted to get a pull request up.

File: Logger/Book.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
+﻿namespace Logger;
+
+public record class Book : BaseEntity
+
+// The Name property is implemented implicitly because the IEntity interface defines it,
+// and we override it by returning the books Title.
+{
+    public string Title { get; }
+
+    public Book(string title)
```
Comment (dpalmer10) [inline Logger/Book.cs:10]: Everything looks great, you could maybe add an author to a book so that it has a little more information available. Good work.

**Review by dpalmer10 — COMMENTED** [review]
Define a full name record (first, last, middle) handling optional and null appropriately. ✔
        Provide a comment on the full name record on why you selected to define a value or a reference type and ✔
        Provide a comment on the full name record on why or why not the type is immutable. ✔
    Define an IEntity interface: ✔
        Add an Id property of type Guid that is an init-only setter. ✔
        Add a Name property that is string. ✔
    Define an abstract base class that implements IEntity - appropriately choosing to implement the interface explicitly or implicitly. ✔
        Do not implement the Name property in this abstract class. ✔
        Do force any derived classes to provide an implementation for Name. ✔
    Define book, student, and employee records - all with entity capabilities. ✔
        Choose a thoughtful approach for the value of Name for each type of entity. In all cases, there should be no backing field for Name (not even one generated by an automatically implemented property) - consider a calculated property. ✔
        Provide a comment on each interface member in each entity explaining why you implemented it implicitly or explicitly. ✔
        You should consider the relationship between Student and Employee and refactor the common code shared between them. ✔
        Given the properties on the full name record, you should consider which entities it makes sense to use it. ✔
        Test that the equality behavior on these entities behaves as expected. ✔
        Provide a comment on each interface method explaining why you implemented it implicitly or explicitly.
    Unit test the Storage class using various entities to ensure that there are no bugs.
    Code review (using TODO: comments if there are any issues - which there may not be) all logger-related classes.)
        Pay special attention to the CreateLogger methods throughout.

Fundamentals

    Be sure you enable:
        Enable Nullability for all projects ✔
        Set LangVersion and the TargetFramework to the latest released versions available (preview versions optional) ✔
        Ensure that you turn on code analysis for all projects(EnableNETAnalyzers) ✔
        Ensure that you turn on EnforceCodeStyleInBuild ✔
        Ensure that you turn on Nullability (Nullable) ✔
    Ensure there are no errors or warnings (including code analysis warnings) ✔
    All of the above should be unit tested ✔
    Choose simplicity over complexity ✔

Extra Credit

    Refactor your project files to put common properties and/or items into a single file. ✔
    Convert ILogger.CreateLogger to a generic method. ❌✔
        Returns the logger classes type that is to be created. ❌✔
        Has a type parameter for the configuration type ❌✔

File: Logger/Employee.cs (hunk 2 above)
Comment (ccornell88) [reply Logger/Employee.cs]: Done!

File: Logger/Student.cs (hunk 3 above)
Comment (ccornell88) [reply Logger/Student.cs]: Done!

File: Logger/Book.cs (hunk 4 above)
Comment (ccornell88) [reply Logger/Book.cs:10]: Thanks I meant to add something. ill get it done!

File: Logger/Book.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
+﻿namespace Logger;
+
+public record class Book : BaseEntity
+{
+    // The Name property is implemented implicitly because the IEntity interface defines it,
```
Comment (KennethWhite) [inline Logger/Book.cs:5]: Nit: Generally put comments above the property they are referring to

File: Logger.Tests/FullNameTests.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        // Act
+        var result = fullName.ToString();
+
+        // Assert
+        Assert.Equal("Chris Cornell", result);
+
+    }
+
+    [Fact]
+    public void FullName_WithBlankMiddleName_ShouldSetMiddleNameToNull()
```
Comment (KennethWhite) [inline Logger.Tests/FullNameTests.cs:48]: Nit: You could have added an additional check where the middle name was passed in as null, though this test is enough to cover that code block on its own.

**Review by KennethWhite — COMMENTED** [review]
Overall great work! Nothing jumped out at me beyond a couple nit picky things. Keep up the good work!

**Comment by Joshua-Lester3** [comment]
Hey Chris, just a heads up that you need to find a partner for future assignments. We deduct points for this.

**Pull request closed without merging**